pkg/landscaper/installations/helper.go

```diff
@@ -157,22 +157,76 @@ func GetDataImport(ctx context.Context, kubeClient client.Client, contextName st
 }
 
 // GetTargetImport fetches the target import from the cluster.
-func GetTargetImport(ctx context.Context, kubeClient client.Client, contextName string, inst *Installation, targetName string) (*dataobjects.Target, *v1.OwnerReference, error) {
+func GetTargetImport(ctx context.Context, kubeClient client.Client, contextName string, inst *Installation, targetName string) (*dataobjects.Target, error) {
 	// get deploy item from current context
 	raw := &lsv1alpha1.Target{}
 	targetName = helper.GenerateDataObjectName(contextName, targetName)
 	if err := kubeClient.Get(ctx, kubernetes.ObjectKey(targetName, inst.Info.Namespace), raw); err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	owner := kubernetes.GetOwner(raw.ObjectMeta)
 	target, err := dataobjects.NewFromTarget(raw)
 	if err != nil {
-		return nil, nil, fmt.Errorf("unable to create internal target for %s: %w", targetName, err)
+		return nil, fmt.Errorf("unable to create internal target for %s: %w", targetName, err)
+	}
+	return target, nil
+}
+
+// GetTargetListImportByNames fetches the target imports from the cluster, based on a list of target names.
+func GetTargetListImportByNames(ctx context.Context, kubeClient client.Client, contextName string, inst *Installation, targetNames []string) (*dataobjects.TargetList, error) {
+	targets := make([]lsv1alpha1.Target, len(targetNames))
+	for i, targetName := range targetNames {
+		// get deploy item from current context
+		raw := &lsv1alpha1.Target{}
+		targetName = helper.GenerateDataObjectName(contextName, targetName)
+		if err := kubeClient.Get(ctx, kubernetes.ObjectKey(targetName, inst.Info.Namespace), raw); err != nil {
+			return nil, err
+		}
+		targets[i] = *raw
+	}
+	targetList, err := dataobjects.NewFromTargetList(targets)
+	if err != nil {
+		return nil, err
+	}
+
+	return targetList, nil
+}
+
+// GetTargetListImportBySelector fetches the target imports from the cluster, based on a label selector.
+func GetTargetListImportBySelector(ctx context.Context, kubeClient client.Client, contextName string, inst *Installation, selector map[string]string) (*dataobjects.TargetList, error) {
+	targets := &lsv1alpha1.TargetList{}
+	// add context to selector
+	selectorWithContext := map[string]string{}
+	for k, v := range selector {
+		selectorWithContext[k] = v
+	}
+	if len(contextName) != 0 {
+		// top-level targets probably don't have an empty context set, so only add the selector if there actually is a context
+		selectorWithContext[lsv1alpha1.DataObjectContextLabel] = contextName
+	}
+	if err := kubeClient.List(ctx, targets, client.MatchingLabels(selectorWithContext), client.InNamespace(inst.Info.Namespace)); err != nil {
+		return nil, err
+	}
```

Comment:
you can also do the have a selector that gets all target in a namespace that have not set that label.

If we would use such a selector we would not have to loop through them ourself

Reply:
Ah, I didn't find out how to configure a "negative" selector and thought this was not possible. Will check again.